pkg/store/postings_codec.go

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+
+package store
+
+import (
+	"bytes"
+
+	"github.com/golang/snappy"
+	"github.com/pkg/errors"
+	"github.com/prometheus/prometheus/tsdb/encoding"
+	"github.com/prometheus/prometheus/tsdb/index"
+)
+
+const (
+	// These headers should not be a prefix of each other.
+	codecHeaderRaw    = "diff+varint+raw"
+	codecHeaderSnappy = "diff+varint+snappy"
+)
+
+// isDiffVarintEncodedPostings returns true, if input looks like it has been encoded by diff+varint(+snappy) codec.
+func isDiffVarintEncodedPostings(input []byte) bool {
+	return bytes.HasPrefix(input, []byte(codecHeaderRaw)) || bytes.HasPrefix(input, []byte(codecHeaderSnappy))
+}
+
+// diffVarintEncode encodes postings into diff+varint representation and optional snappy compression.
+func diffVarintEncode(p index.Postings, useSnappy bool) ([]byte, error) {
+	buf := encoding.Encbuf{}
+
+	// If we're returning raw data, write the header to the buffer, and then return buffer directly.
```

Comment:
I don't understand this comment really.. maybe placement is not correct? let's remove it, I totally understand this function without this.

Reply:
ok